### PR TITLE
Automate testcase TC25: Verify if the data value secret is created for a package with inline config

### LINKED ISSUE #2479: TC25: Verify if the data value secret is created for a package with inline config

### PR DESCRIPTION
**Pull request by shivaani0505**
### What this PR does / why we need it
Automate testcase TC25: Verify if the data value secret is created for a package with inline config

### Which issue(s) this PR fixes

Fixes #2479 

### Release note

```release-note
Automate create/delete CC Guest cluster for a package with inline config on supervisor using Custom ClusterBootstrap and verify if the data value secret is created successfully on both supervisor and Guest cluster
```

### CODE REVIEW

File: addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go [hunk 1]
```diff
@@ -368,6 +368,8 @@ func (h *Helper) HandleExistingClusterBootstrap(clusterBootstrap *runtanzuv1alph
 					}
 				}
 			}
+			h.Logger.Info("Updating packagesToBeCloned with additional packages")
+			packagesToBeCloned = append(packagesToBeCloned, clusterBootstrap.Spec.AdditionalPackages...)
```
Comment (maralavi) [inline addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go:372]: I think the right way is to add the following block to line 344 https://github.com/vmware-tanzu/tanzu-framework/blob/main/addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go#L344

```
var nonNilValuesFromPackages []*runtanzuv1alpha3.ClusterBootstrapPackage
			for _, cbPackage := range packages {
				if cbPackage != nil && cbPackage.ValuesFrom != nil {
					nonNilValuesFromPackages = append(nonNilValuesFromPackages, cbPackage)
				}
			}
```

And the do the following instead in line 371 above:
```
packagesToBeCloned = append(packagesToBeCloned, nonNilValuesFromPackages...)
```

File: addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go [hunk 2]
```diff
@@ -368,6 +375,8 @@ func (h *Helper) HandleExistingClusterBootstrap(clusterBootstrap *runtanzuv1alph
 					}
 				}
 			}
+			h.Logger.Info("Updating packagesToBeCloned with additional packages")
```
Comment (maralavi) [inline addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go:378]:
```suggestion
h.Logger.Info("Updating packagesToBeCloned with inline packages")
```

File: addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go (hunk 1 above)
Comment (shivaani0505) [reply addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go]: used the suggested fix

File: addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go (hunk 2 above)
Comment (shivaani0505) [reply addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go]: Fixed

File: pkg/v1/tkg/test/tkgctl/tkgs_cc/tkgs_cc_workload_cluster_test.go [hunk 3]
```diff
@@ -170,6 +171,45 @@ var _ = Describe("TKGS ClusterClass based workload cluster tests", func() {
 			shared.TestClusterUpgrade(tkgctlClient, clusterName, namespace)
 			shared.CheckTKGSAddons(context.TODO(), tkgctlClient, svClusterName, clusterName, namespace, e2eConfig.TKGSKubeconfigPath, e2eConfig.InfrastructureName, true)
 		})
+
+		It("should return success or error based on the ClusterClass with inline package", func() {
```
Comment (maralavi) [inline pkg/v1/tkg/test/tkgctl/tkgs_cc/tkgs_cc_workload_cluster_test.go:175]: Nit: The description could be improved to something in the following lines:
```should create the data value secret in supervisor and guest cluster for a package with inline config```

**Comment by maralavi** [comment]
@shivaani0505 The changes look good. Just the linter issue needs to be resolved and left a minor comment on the improvement of the description for the testcase

File: pkg/v1/tkg/test/tkgctl/tkgs_cc/tkgs_cc_workload_cluster_test.go (hunk 3 above)
Comment (shivaani0505) [reply pkg/v1/tkg/test/tkgctl/tkgs_cc/tkgs_cc_workload_cluster_test.go]: Fixed

File: pkg/v1/tkg/test/data/testdata/customClusterBootstrapInline.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        value: best-effort-small
+      - name: storageClass
+        value: wcpglobalstorageprofile
+    version: v1.23.5+vmware.1-tkg.1-zshippable
+    workers:
+      machineDeployments:
+        - class: node-pool
+          metadata: {}
+          name: np-2
+          replicas: 1
```
Comment (maralavi) [inline pkg/v1/tkg/test/data/testdata/customClusterBootstrapInline.yaml:61]: @shivaani0505 Pipeline's linter would complain again if you don't add a new line here

Comment (shivaani0505) [reply pkg/v1/tkg/test/data/testdata/customClusterBootstrapInline.yaml]: Fixed

**Review by maralavi — APPROVED** [review]

**Pull request merged**